packages/SwingSet/src/vats/comms/delivery.js

```diff
@@ -125,41 +141,56 @@ export function makeDeliveryKit(state, syscall, transmit, clistKit, stateKit) {
   }
 
   function resolveFromRemote(remoteID, message) {
-    const sci = message.indexOf(';');
-    assert(sci !== -1, details`missing semicolon in resolve ${message}`);
-    // message is created by resolveToRemote, so one of:
-    // `resolve:object:${target}:${resolutionRef};`
-    // `resolve:data:${target}${rmss};${resolution.body}`
-    // `resolve:reject:${target}${rmss};${resolution.body}`
-
-    const pieces = message.slice(0, sci).split(':');
-    // pieces[0] is 'resolve'
-    const type = pieces[1];
-    const remoteTarget = pieces[2];
-    const remoteSlots = pieces.slice(3); // length=1 for resolve:object
-    insistRemoteType('promise', remoteTarget); // slots[0] is 'rp+NN`.
-    const vpid = getLocalForRemote(remoteID, remoteTarget);
-    // rp+NN maps to target=p-+NN and we look at the promiseTable to make
-    // sure it's in the right state.
-    insistPromiseIsUnresolved(vpid);
-    insistDeciderIsRemote(vpid, remoteID);
-
-    const slots = remoteSlots.map(s => provideLocalForRemote(remoteID, s));
-    const body = message.slice(sci + 1);
-    const data = harden({ body, slots });
-    changeDeciderFromRemoteToComms(vpid, remoteID);
+    // message is created by resolveToRemote.  It consists of 1 or more
+    // resolutions, separated by newlines, each taking the form:
+    // `resolve:${rejected}:${target}${rmss};${resolution.body}`
```

Comment:
minor nit, since we're no longer using `true`/`false` for `${rejected}`, and since this comment serves as protocol documentation, let's expand the comment here:

```js
// resolutions, separated by newlines, each taking the form of either:
// `resolve:reject:${target}${rmss};${resolution.body}`
// or
// `resolve:fulfill:${target}${rmss};${resolution.body}`
```

Reply:
done